src/web/src/main/java/tupperdate/web/facade/chats/ChatFacadeImpl.kt

```diff
@@ -1,38 +1,146 @@
 
 class ChatFacadeImpl(
-    private val chats: ChatRepository,
+    private val chatRepository: ChatRepository,
+    private val userRepository: UserRepository,
+    private val recipeRepository: RecipeRepository,
+    private val notificationRepository: NotificationRepository,
 ) : ChatFacade {
 
     override suspend fun readAll(
         user: User,
     ): Result<List<Chat>> {
-        TODO("Not yet implemented")
+        val convs: List<Conversation> = when (val res = chatRepository.readAll(user)) {
+            is Result.Ok -> res.result
+            else -> return res.map { listOf() }
+        }
+
+        val chats = convs.map { conv ->
+            val chatUser =
+                when (val res = userRepository.read(user = User(AccountId(conv.theirId)))) {
+                    is Result.Ok -> res.result
+                    else -> return res.map { listOf() }
+                }
+            val lastMessage =
+                when (val res = chatRepository.readMessages(user = user, userId = conv.theirId)) {
+                    is Result.Ok -> res.result
+                    else -> return res.map { listOf() }
+                }.getOrNull(0)
+            val myRecipes = conv.myRecipes.map {
+                when (val res = recipeRepository.readOne(user, it)) {
+                    is Result.Ok -> res.result
+                    else -> return res.map { listOf() }
+                }
+            }
+            val theirRecipes = conv.theirRecipes.map {
+                when (val res = recipeRepository.readOne(User(AccountId(conv.theirId)), it)) {
+                    is Result.Ok -> res.result
+                    else -> return res.map { listOf() }
+                }
+            }
+
+            return@map Chat(
+                userId = conv.theirId,
+                displayName = chatUser.displayName,
+                picture = chatUser.displayPicture?.url,
+                lastMessage = lastMessage?.toMessage(),
+                myRecipes = myRecipes.map { it.toRecipe() },
+                theirRecipes = theirRecipes.map { it.toRecipe() },
+            )
+
+        }
+
+        return Result.Ok(chats)
     }
 
     override suspend fun readOne(
         user: User,
+        userId: String,
     ): Result<Chat> {
-        TODO("Not yet implemented")
+        val conv: Conversation = when (val res = chatRepository.readOne(user, userId = userId)) {
+            is Result.Ok -> res.result
+            else -> return res.map { emptyChat() }
+        }
+
+        val chatUser = when (val res = userRepository.read(user = User(AccountId(conv.theirId)))) {
+            is Result.Ok -> res.result
+            else -> return res.map { emptyChat() }
+        }
+        val lastMessage =
+            when (val res = chatRepository.readMessages(user = user, userId = conv.theirId)) {
+                is Result.Ok -> res.result
+                else -> return res.map { emptyChat() }
+            }.getOrNull(0)
```

Comment:
We probably don't want to scan through the whole message set of a user each time, but rather only take the latest directly.